### PR TITLE
bug fix - login after can't access to setting page

### PR DESCRIPTION
**Pull request by wiroger9595**
### What this PR does (required):
- fix profile get null when user login

### Screenshots / Videos (front-end only):
before
Screen Shot 2022-01-23 at 7 01 05 PM

after
Screen Shot 2022-01-23 at 7 01 33 PM

### Any information needed to test this feature (required):
- you can see there is no Profile object after user login because the website can't get the object profile, so the website automatically back to the login page, so make sure any page you want to login website able to get the object they want

### CODE REVIEW

**Review by MUGABA — COMMENTED** [review]
Hey! @wiroger9595, I was the login was not passing the profile data that's why it was null, Smart what you did here!

File: server/controllers/auth.js [hunk 1]
```diff
@@ -107,10 +110,15 @@ exports.loadUser = asyncHandler(async (req, res, next) => {
   const user = await User.findById(req.user.id);
   const profile = await Profile.findOne({ userId: req.user.id });
 
+  console.log("use loaduser");
```
Comment (MUGABA) [inline server/controllers/auth.js:113]: would you mind removing these logs!

File: client/src/components/AvatarDisplay/AvatarDisplay.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ interface Props {
 }
 
 const AvatarDisplay = ({ user }: Props): JSX.Element => {
+  console.log('avatar display here');
```
Comment (MUGABA) [inline client/src/components/AvatarDisplay/AvatarDisplay.tsx:10]: @moffatethan might not like, leaving bug logs and comments in the code. If you could try removing them before you push

**Review by MUGABA — CHANGES_REQUESTED** [review]
Let leave this, for you it may help! If you can access the profile data on your side, make sure you do! because in the `Settings Page` It will redirect you to login 
```
const { loggedInUser, profile } = useAuth();
  const history = useHistory();
  const classes = useStyles();

  if (loggedInUser === undefined) return <CircularProgress />;
  if (!loggedInUser || !profile) {
    history.push('/login');
    // loading for a split seconds until history.push works
    return <CircularProgress />;
  }
```

And in if its possible could you work with `manage-bookings-ui ` branch it has updated routing merged to it!

**Comment by wiroger9595** [comment]
`const { loggedInUser, profile } = useAuth();
  const history = useHistory();
  const classes = useStyles();

  if (loggedInUser === undefined) return <CircularProgress />;
  if (!loggedInUser || !profile) {
    history.push('/login');
    // loading for a split seconds until history.push works
    return <CircularProgress />;
  }`

there is no change on this code

File: server/controllers/auth.js [hunk 3]
```diff
@@ -92,6 +93,8 @@ exports.loginUser = asyncHandler(async (req, res, next) => {
           name: user.name,
           email: user.email,
         },
+        profile,
+
```
Comment (MUGABA) [inline server/controllers/auth.js:97]: Look the user is already loaded and sent to using the endpoint `/auth/user` and at the ui it's fetched using the function `loginWithCookies`.  the fix was in my opinion is make sure its visible by the user through `/profile/settings`. Would mind thinking about this from the ui's side not the back

**Comment by MUGABA** [comment]
> Let leave this, for you it may help! If you can access the profile data on your side, make sure you do! because in the `Settings Page` It will redirect you to login
> 
> ```
> const { loggedInUser, profile } = useAuth();
>   const history = useHistory();
>   const classes = useStyles();
> 
>   if (loggedInUser === undefined) return <CircularProgress />;
>   if (!loggedInUser || !profile) {
>     history.push('/login');
>     // loading for a split seconds until history.push works
>     return <CircularProgress />;
>   }
> ```
> 
> And in if its possible could you work with `manage-bookings-ui ` branch it has updated routing merged to it!

File: server/controllers/auth.js [hunk 4]
```diff
@@ -92,6 +93,8 @@ exports.loginUser = asyncHandler(async (req, res, next) => {
           name: user.name,
           email: user.email,
         },
+        profile,
```
Comment (MUGABA) [inline server/controllers/auth.js:96]: Ya this will do it! lets see what ethan will say!

Comment (moffatethan) [reply server/controllers/auth.js:96]: This is great! Thanks!

**Review by moffatethan — APPROVED** [review]

**Comment by wiroger9595** [comment]
merge null point exception

**Pull request merged**